### PR TITLE
Sort option changes

### PR DESCRIPTION
**Pull request by JuanAlmeida12**
#### What is the purpose of this pull request?

Change sort option to dropdown button in search-result

#### What problem is this solving?

Sort option in search-result is redesign to be  a dropdown button

#### How should this be manually tested?
[Access this workspace](https://orderby--storecomponents.myvtex.com/veiculos/d)
#### Screenshots or example usage
orderby--storecomponents myvtex com_veiculos_d iphone 6_7_8 plus

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [x] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/global.css [hunk 1]
```diff
@@ -213,3 +213,57 @@ body.vtex-filter-popup-open {
     grid-template-columns: repeat(2, 1fr);
   }
 }
+
```
Comment (ArthurSampaio) [inline react/global.css]: Given a look in the css code, i think that some attributes can be replaced with design tokens, like the borders, colors and grid markers. This should improve the stylization of the component. What do you think?

File: react/components/SelectionListOrderBy.js [hunk 2]
```diff
@@ -22,9 +29,16 @@ class SelectionListOrderBy extends Component {
     }),
   }
 
-  handleSelect = option => e => {
-    e.preventDefault()
+  renderOptions = () => {
+    const { options } = this.props
+    return options.map(option => {
+      return (
+        <a href onClick={() => this.handleSelect(option)} key={option.value}>{option.label}</a>
```
Comment (brunojdo) [inline react/components/SelectionListOrderBy.js]: Use navigate

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,7 +6,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-### Added
+### Changed
+- Sort option in search-result is redesign to be dropdown button
+
+### Changed
```
Comment (lucasecdb) [inline CHANGELOG.md]: `Added`?

File: react/components/Popup.js [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
+
+
```
Comment (lucasecdb) [inline react/components/Popup.js]: remove ⛏

File: react/components/Popup.js [hunk 5]
```diff
@@ -95,7 +97,7 @@ export default class Popup extends Component {
           })
 
           const contentClassName = classNames(
-            'vtex-filter-popup__content-container h-auto bg-base fixed dn w-100 left-0 bottom-0 z-1 ph3 overflow-y-auto flex-column',
+            'vtex-filter-popup__content-container h-auto bg-white fixed dn w-100 left-0 bottom-0 z-1 ph3 overflow-y-auto flex-column',
```
Comment (lucasecdb) [inline react/components/Popup.js]: why remove the design token?

File: react/components/Popup.js [hunk 6]
```diff
@@ -116,15 +118,15 @@ export default class Popup extends Component {
             <div className={className} ref={this.contentRef}>
             
               <button
-                className={classNames('vtex-filter-popup__button mv0 pointer flex justify-center items-center', {
-                  'bb b--muted-1': open,
+                className={classNames('vtex-filter-popup__button pa5 mv0 mh5 pointer flex justify-center items-center', {
+                  'bb b--dark-gray': open,
```
Comment (lucasecdb) [inline react/components/Popup.js]: same as above

File: react/components/Popup.js [hunk 7]
```diff
@@ -116,15 +118,15 @@ export default class Popup extends Component {
...
                   'bn': !open,
                 })}
                 onClick={onToggle(id)}
               >
-                <span className="vtex-filter-popup__title c-on-base t-action--small ml-auto">{title}</span>
-                <span className="vtex-filter-popup__arrow-icon ml-auto pl3 pt2">
-                  {icon}
+                <span className="vtex-filter-popup__title f5 ml-auto">{title}</span>
```
Comment (lucasecdb) [inline react/components/Popup.js]: same

File: react/components/SearchResult.js [hunk 8]
```diff
@@ -35,7 +35,7 @@ class SearchResult extends Component {
 
   handleLayoutChange = (e, mode) => {
     e.preventDefault()
-
+
```
Comment (lucasecdb) [inline react/components/SearchResult.js]: remove ⛏

File: react/components/SelectionListOrderBy.js (hunk 2 above)
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: Can't you use the `Link` component from render?

File: react/components/SelectionListOrderBy.js [hunk 9]
```diff
@@ -37,39 +51,34 @@ class SelectionListOrderBy extends Component {
     })
   }
 
+  getOrderByTitle = order => {
```
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: `getOptionTitle` ⛏

also rename the parameter to `option`

File: react/components/SelectionListOrderBy.js [hunk 10]
```diff
@@ -37,39 +51,34 @@ class SelectionListOrderBy extends Component {
...
-
-                {active && (
-                  <span className="vtex-orderby__item-icon fr">
-                    <CheckTick />
-                  </span>
-                )}
-              </div>
-            )
-          })}
+      <div align="center|justify" className="vtex-dropdown flex justify-center flex-auto pt1">
```
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: what's this `align` property?

File: react/components/SelectionListOrderBy.js [hunk 11]
```diff
@@ -37,39 +51,34 @@ class SelectionListOrderBy extends Component {
...
+      })
+    }
+    return title
+  }
+
   render() {
-    const { intl, options, orderBy } = this.props
+    const { orderBy } = this.props
+    const { showDropdown } = this.state
+    const btClass = `bg-base c-on-base t-action--small ml-auto ${showDropdown ? 'vtex-dropdown-btn__active b--muted-4' : 'vtex-dropdown-btn'}`
```
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: the `vtex-dropdown-btn` class should always be present.

use the `classnames` package.

Comment (lucasecdb) [reply react/components/SelectionListOrderBy.js]: also, rename it to `vtex-dropdown__btn`, follow the [BEM](https://en.bem.info/methodology/quick-start/#element) best practices.

File: react/components/SelectionListOrderBy.js [hunk 12]
```diff
@@ -37,39 +51,34 @@ class SelectionListOrderBy extends Component {
...
-                )}
-              </div>
-            )
-          })}
+      <div align="center|justify" className="vtex-dropdown flex justify-center flex-auto pt1">
+        <button onClick={this.handleDropdownBtClick} className={btClass}>
+          {this.getOrderByTitle(orderBy)}
+          <span className="vtex-orderby__item-icon fr v-mid"><Arrow size={16} /></span>
+        </button>
+        <div className={`vtex-dropdown-content bg-base ${showDropdown ? 'b--muted-4 vtex-dropdown-content__active' : ''}`}>
```
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: rename the `vtex-dropdown-content` to `vtex-dropdown__content`

and rename the `vtex-dropdown-content__active` to `vtex-dropdown__content--active`

File: react/components/SelectionListOrderBy.js (hunk 11 above)
Comment (lucasecdb) [reply react/components/SelectionListOrderBy.js]: and the `vtex-dropdown-btn__active` to `vtex-dropdown__btn--active`

File: react/components/SelectionListOrderBy.js (hunk 10 above)
Comment (luciannojunior) [reply react/components/SelectionListOrderBy.js]: @lucasecdb is html's native property for alignment. Should be avoided in favor of CSS

File: CHANGELOG.md (hunk 3 above)
Comment (JuanAlmeida12) [reply CHANGELOG.md]: `Added` is removed.

Comment (lucasecdb) [reply CHANGELOG.md]: what? but there's 2 `Changed` in the changelog, and before your commit, the changes bellow the line I commented were inside an `### Added` block

File: react/components/Popup.js [hunk 13]
```diff
@@ -116,14 +118,14 @@ export default class Popup extends Component {
...
-                className={classNames('vtex-filter-popup__button mv0 pointer flex justify-center items-center', {
+                className={classNames('vtex-filter-popup__button pa5 mv0 mh5 pointer flex justify-center items-center', {
                   'bb b--muted-1': open,
                   'bn': !open,
                 })}
                 onClick={onToggle(id)}
               >
                 <span className="vtex-filter-popup__title c-on-base t-action--small ml-auto">{title}</span>
-                <span className="vtex-filter-popup__arrow-icon ml-auto pl3 pt2">
+                <span className="vtex-filter-popup__arrow-icon ml-auto  pl3 pt2">
```
Comment (luciannojunior) [inline react/components/Popup.js]: Lint or pretty your code. There is an extra space between the classes

File: react/global.css [hunk 14]
```diff
@@ -115,6 +115,14 @@
   width: 2px;
 }
 
+.vtex-search-result__border-2 {
+  grid-column: borderseq;
+  grid-row: borderseq;
+  -ms-grid-column: 2;
+  -ms-grid-row: 2;
+  width: 2px;
```
Comment (luciannojunior) [inline react/global.css]: Are all pixel values in this value really necessary? `rem` use is encouraged

File: react/global.css [hunk 15]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
     grid-template-columns: repeat(2, 1fr);
   }
 }
+
+.vtex-dropdown-btn {
+  padding: 16px 12px 11px 8px;
+  font-size: 16px;
+  width: 100%;
```
Comment (lucasecdb) [inline react/global.css]: use class `w-100`

File: react/global.css [hunk 16]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
...
+  border: none;
```
Comment (lucasecdb) [inline react/global.css]: class `bn`

File: react/global.css [hunk 17]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
...
+
+.vtex-dropdown-btn:focus {
+  outline: none;
+}
+
+.vtex-dropdown {
+  width: 100%;
+  margin-top: -3px;
+  position: relative;
+  display: inline-block;
```
Comment (lucasecdb) [inline react/global.css]: class `dib`

File: react/global.css [hunk 18]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
...
+}
+
+.vtex-dropdown-content {
+  display: none;
```
Comment (lucasecdb) [inline react/global.css]: class `dn`

File: react/global.css [hunk 19]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
...
+  position: absolute;
+  width: 100%;
```
Comment (lucasecdb) [inline react/global.css]: class `w-100`

File: react/global.css [hunk 20]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
...
+  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
+  z-index: 1;
```
Comment (lucasecdb) [inline react/global.css]: class `z-1`

File: react/global.css [hunk 21]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
...
+}
+
+.vtex-dropdown-content a {
+  color: black;
```
Comment (lucasecdb) [inline react/global.css]: `c-on-base`

File: react/global.css [hunk 22]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
...
+  padding: 12px 16px;
+  max-width: 100%;
+  text-decoration: none;
```
Comment (lucasecdb) [inline react/global.css]: `no-underline`

File: react/global.css [hunk 23]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
...
+  display: block;
+}
+
+.vtex-dropdown-content a:hover {background-color: #f1f1f1}
+
+.vtex-dropdown-content__active {
+  display: block;
```
Comment (lucasecdb) [inline react/global.css]: `db`

File: react/global.css [hunk 24]
```diff
@@ -213,3 +221,57 @@ body.vtex-filter-popup-open {
...
+  border-style: solid;
+  border-width: 2px 2px 2px 2px;
+  border-radius: 0px 0px 4px 4px;
+}
+
+.vtex-dropdown-btn__active {
+  padding: 16px 12px 11px 8px;
+  font-size: 16px;
```
Comment (lucasecdb) [inline react/global.css]: `f5`

File: react/components/SelectionListOrderBy.js [hunk 25]
```diff
@@ -22,54 +30,56 @@ class SelectionListOrderBy extends Component {
     }),
   }
 
-  handleSelect = option => e => {
-    e.preventDefault()
+  renderOptions = () => {
+    const { options } = this.props
+    return options.map(option => {
+      return (
+        <Link {...this.getLinkProp(option.value)}><a>{option.label}</a></Link>
```
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: Remove the `a` tag

File: react/components/SelectionListOrderBy.js [hunk 26]
```diff
@@ -22,54 +30,63 @@ class SelectionListOrderBy extends Component {
...
+  }
 
-    navigate({
-      page: linkProps.page,
-      query: linkProps.queryString,
-      params: linkProps.params,
-      scrollOptions: { baseElementId: 'search-result-anchor', top: -HEADER_SCROLL_OFFSET },
+  getOptionTitle = option => {
+    const { options, intl } = this.props
+    console.log(option)
```
Comment (ArthurSampaio) [inline react/components/SelectionListOrderBy.js]: remove

**Comment by luciannojunior** [comment]
image

The :hover is one time working and other not:

image

In one it gets with this weird shape, and in the other it gets normal, but without the `cursor: pointer`.

Besides that, the names are not centered in the list

File: CHANGELOG.md [hunk 27]
```diff
@@ -14,12 +14,13 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Added
 - Add new mobile design.
```
Comment (brunojdo) [inline CHANGELOG.md]: Move to Unreleased

File: react/components/SelectionListOrderBy.js [hunk 28]
```diff
@@ -22,54 +30,58 @@ class SelectionListOrderBy extends Component {
...
-        title={intl.formatMessage({ id: 'search-result.orderby.title' })}
-        id="orderby"
-      >
-        <div className="vtex-orderby-popup">
-          {options.map(opt => {
-            const active = orderBy === opt.value
+  render() {
+    const { orderBy } = this.props
+    const { showDropdown } = this.state
+    const btClass = classNames('ph3 pv5 mv0 pointer flex justify-center items-center bn w-100 bg-base c-on-base t-action--small ml-auto',
```
[the remark this answers is not in the document]
Comment (JuanAlmeida12) [reply react/components/SelectionListOrderBy.js]: I remove this class `vtex-dropdown__btn`, i use `Design Token` and `Tachyons`

**Pull request merged**